README.md

```diff
@@ -433,6 +433,203 @@ to confirm that our debugging object is still displayed:
+- **`Fetch.fetch(url)`**: Initiates a GET request to the specified URL.
+- **`then_(Fetch.Response.text)`**: Converts the response into a text string.
+- **`then_(text => Js.log(text) |> resolve)`**: Logs the response text to the
+  console.
+- **`ignore`**:
+  [Discards](https://reasonml.github.io/api/Stdlib.html#1_Unitoperations) the
+  final promise result since we're only interested in side effects.
+
+> [!TIP]
+> 
+> Chaining promises with `then_` allows you to handle asynchronous code in a
+> structured way, similar to how you'd do it in JavaScript. We will see later on
+> a different way to handle asynchronous code, that is closer to `async/await`.
+
+### 6.4: Step 6 completion check
+
+Run your application to ensure that the data is being fetched correctly. Check
+your console to see the printed data. If everything is working, you should see
+the raw JSON data in your console.
+
+If everything works well, now it's a good time to clean up some of the previous
+prototyping code that we are not using anymore: `Feed.data`, `Feed.demoFeed` and
+the `Js.log(Feed.feed_of_json);` line at the end of `App.re`.
+
+## Step 7: Displaying data
+
+Now that we have successfully fetched the data, let’s move on to decoding it and
+displaying it in the UI.
+
+### 7.1: Decoding the fetched data
+
+We need to decode the raw data using our `feed_of_json` function. First of all,
+we are going to define a new variant type to specify the state of the UI:
+
+```reason
+type loadingStatus =
+  | Loading
+  | Loaded(Result.t(Feed.feed, string));
+```
+
+This type uses a few things:
+- [Variant types](https://reasonml.github.io/docs/en/variant), similar to enums
+- Result type, which is a predefined variant with two options: `Ok` and `Error`
+- In this case, the "Ok" part is the `feed` type we defined in the previous
+  step, and the error type is just a string
+
+Now we can adapt the effect in `App.re` to use the new type. We will also be
+adding a `useState` hook to store the data loading state:
+
+
+```reason
+/* inside the App component `make` function */
+    let (data, setData) = React.useState(() => Loading);
+    React.useEffect0(() => {
+      Js.Promise.(
+        Fetch.fetch("https://gh-feed.vercel.app/api?user=jchavarri&page=1")
+        |> then_(Fetch.Response.text)
+        |> then_(text =>
+             {
+               let data =
+                 try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                 | Json.Decode.DecodeError(msg) =>
+                   Js.Console.error(msg);
+                   Error("Failed to decode: " ++ msg);
+                 };
+               setData(_ => Loaded(data));
+             }
+             |> resolve
+           )
+      )
+      |> ignore;
+      None;
+    });
+```
+
+### 7.2: Rendering the data in the UI
+
+With the data decoded, it’s time to display it in the UI. Update your React
+component to process the result and render the feed:
+
+```reason
```

Comment:
When I paste this in, the styles in `index.html` make the top part unreadable:
![image](https://github.com/user-attachments/assets/4c672d05-39ff-43a3-97f5-fa42e07a9825)

Reply:
Yeah the styles at the moment are 💩 . I am talking to @davesnx to figure out the best way to do them (use styled-ppx, GitHub style system...). Maybe we should start without any styles and build them later on. In any case, I will apply any styles fixes retroactively to this and other steps when the decision is clear.

Reply:
One way to do it is to keep styles simple, show how to style a small part using styled-ppx and then let them style the rest of it themselves as an optional exercise